lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+          issue.add_comment("@#{issuer} '#{value}' is an invalid repo, ignoring...")
+
+          return false
+        end
+
+        true
+      end
+
+      REPO_ONLY_REGEXP = /(?:[^\/]+\/)?(?<REPO_NAME_ONLY>[^@#]+)/.freeze
+      def repos_valid?(repos)
+        repos.all? do |repo|
+          name_only = repo.match(REPO_ONLY_REGEXP)[:REPO_NAME_ONLY]
+          GithubService.org_repos(issue.organization_name).include?(name_only)
+        end
+      end
+
+      ##### run_tests steps #####
+
+      # Clone repo (if needed) and initialize @rugged_repo
+      def clone_repo_if_necessary
+        repo_path = self.class.test_repo_clone_dir
+        if Dir.exist?(self.class.test_repo_clone_dir)
+          @rugged_repo = Rugged::Repository.new(repo_path)
+        else
+          url = self.class.test_repo_url
+          @rugged_repo = Rugged::Repository.clone_at(url, repo_path)
+        end
+        git_fetch
+      end
+
+      def create_cross_repo_test_branch
+        rugged_repo.create_branch(branch_name, "origin/master")
+        rugged_repo.checkout(branch_name)
+      end
+
+      def update_travis_yaml_content
+        Dir.chdir(@rugged_repo.workdir) do
+          content = YAML.load_file(".travis.yml")
+
+          content["env"] = {} unless content["env"]
+          content["env"]["global"] = ["REPOS=#{repos.join(',')}"]
+          content["env"]["matrix"] = test_repos.map { |repo| "TEST_REPO=#{repo}" }
+
+          File.write('.travis.yml', content.to_yaml)
+        end
+      end
+
+      def commit_travis_yaml_changes
+        index = rugged_repo.index
+        index.add('.travis.yml')
+        index.write
+
+        # rubocop:disable: Rails/TimeZone
+        bot       = self.class.bot_name
+        author    = { :name => issuer, :email => "no-name@example.com", :time => Time.now }
+        committer = { :name => bot,    :email => "#{bot}@manageiq.org", :time => Time.now }
+        # rubocop:enable: Rails/TimeZone
+
+        Rugged::Commit.create(
+          rugged_repo,
+          :author     => author,
+          :committer  => committer,
+          :parents    => [rugged_repo.last_commit].compact,
+          :tree       => index.write_tree(rugged_repo),
+          :update_ref => "HEAD",
+          :message    => <<~COMMIT_MSG
+            Running tests for #{issuer}
+
+            From Pull Request:  #{issue.fq_repo_name}##{issue.number}
+          COMMIT_MSG
+        )
+      end
+
+      def push_commit_to_remote
+        push_options = {}
+
+        if Settings.github_credentials.username && Settings.github_credentials.password
+          rugged_creds = Rugged::Credentials::UserPassword.new(
+            :username => Settings.github_credentials.username,
+            :password => Settings.github_credentials.password
+          )
+          push_options[:credentials] = rugged_creds
+        end
+
+        remote = @rugged_repo.remotes['origin']
+        remote.push(["refs/heads/#{branch_name}"], push_options)
```

Comment:
We may have to be careful if the branch_name has a `/` in it...particularly if we insert the org/repo_name into the branch name as per other comments on the branch name structure.

Reply:
Yeah, that is a good point.  I will look into this.  Remind me if I forget after addressing some things, since this is important.

Reply:
I just plan on inserting `issue.repo_name` for what it is worth, since `org` should be assumed since we restrict by that anyway.

Reply:
I think this is mostly address as is.